server/src/main/java/org/elasticsearch/index/mapper/SourceValueFetcher.java

```diff
@@ -63,7 +64,7 @@ public SourceValueFetcher(String fieldName, MapperService mapperService, boolean
             }
 
             if (parsesArrayValue) {
-                values.addAll((List<?>) parseSourceValue(sourceValue));
+                values.addAll((List<?>) tryParseSourceValue(sourceValue));
```

Comment:
If this comes back null I think it'll blow up.

Reply:
This is a good point -- I assumed an implicit contract with the array value mappers where they would return an empty list instead of `null`. I'll add a null check here though for safety.